src/Microsoft.ApplicationInsights.AspNetCore/Extensibility/Implementation/Tracing/AspNetEventSource.cs

```diff
@@ -134,6 +134,12 @@ public void FetchAppIdFailedWithResponseCode(string exception, string appDomainN
             this.WriteEvent(10, exception, this.ApplicationName);
         }
 
+        [Event(11, Message = "Warning Message: {0}", Level = EventLevel.Warning, Keywords = Keywords.Diagnostics)]
+        public void LogWarningMessage(string message, string appDomainName = "Incorrect")
```

Comment:
Not sure this is a common pattern we use.

Reply:
Maybe you can use ILogger if loggin in this scenario is required.

Reply:
All other AI sdk uses EventSource to report errors like this.